src/database/database.py

```diff
@@ -1897,3 +1884,231 @@ def get_friend_requests_for_target_user(self, target: str):
+
+        Args:
+            recipes (list): The list of recipes to add. Each recipe must be a JSON object containing the following values:
+                id (int): The ID of the recipe.
+                name (str): The name of the recipe.
+                image (str): The image URL of the recipe.
+                summary (str): The summary of the recipe. This value is optional.
+                full_summary (str): The full summary of the recipe. This value is optional.
+        """
+
+        from database.models import Recipe
+
+        actual_recipes = []
+
+        for recipe in recipes:
+            if self.get_recipe(recipe["id"]) != None:
+                raise DuplicateRecipeException(recipe["id"])
+            summary = ""
+            full_summary = ""
+            if "summary" in recipe.keys():
+                summary = recipe["summary"]
+            if "full_summary" in recipe.keys():
+                full_summary = recipe["full_summary"]
+
+            actual_recipes.append(
+                Recipe(
+                    id=recipe["id"],
+                    name=recipe["name"],
+                    image=recipe["image"],
+                    summary=summary,
+                    full_summary=full_summary,
+                )
+            )
+
+        session = self.int__Session()
+        try:
+            session.add_all(actual_recipes)
+            session.commit()
+        except:
+            session.rollback()
+            raise DatabaseException("Failed to add recipes")
+        finally:
+            session.close()
+
+    def add_ingredients(self, ingredients: list):
+        """
+        This is a convenience function for adding ingredients in bulk.
+        This is the same as making repeated calls to `add_ingredient`, but this function is faster.
+
+        Args:
+            ingredients (list): The list of ingredients to add. Each ingredient must be a JSON object containing the following values:
+                id (int): The ID of the ingredient.
+                name (str): The name of the ingredient.
+                image (str): The image URL of the ingredient.
+        """
+
+        from database.models import Ingredient
+
+        actual_ingredients = []
+
+        for ingredient in ingredients:
+            if self.get_ingredient(ingredient["id"]) != None:
+                raise DuplicateIngredientException(ingredient["id"])
+
+            actual_ingredients.append(
+                Ingredient(
+                    id=ingredient["id"],
+                    name=ingredient["name"],
+                    image=ingredient["image"],
+                )
+            )
+
+        session = self.int__Session()
+        try:
+            session.add_all(actual_ingredients)
+            session.commit()
+        except:
+            session.rollback()
+            raise DatabaseException("Failed to add ingredients")
+        finally:
+            session.close()
+
+    def add_google_users(self, users: list):
```

Comment:
I suggest moving this database function below and group it with all user authentication functions, while keeping all recipe and ingredient functions in a cluster